MediaBrowser.Api/Playback/Hls/BaseHlsService.cs

```diff
@@ -209,13 +209,14 @@ protected virtual async Task WaitForMinimumSegmentCount(string playlist, int seg
                 try
                 {
                     // Need to use FileShare.ReadWrite because we're reading the file at the same time it's being written
-                    using var fileStream = GetPlaylistFileStream(playlist);
+                    await using var fileStream = GetPlaylistFileStream(playlist);
+                    fileStream.ConfigureAwait(false);
```

Comment:
I'm not 100% sure, but I think you want to do this instead. Otherwise you are awaiting the un-configured instance

```cs
var fileStream = GetPlaylistFileStream(playlist);
await using fileStream.ConfigureAwait(false);
```

I couldn't find any examples of doing it the way you have it, so I'm not sure if it is equivalent. For example, I found this which only has example of doing it my way: https://github.com/dotnet/csharplang/issues/2661

Reply:
That doesn't compile
```cs
Playback/Hls/BaseHlsService.cs(213,58): error CS1001: Identifier expected [/home/bond/dev/jellyfin/MediaBrowser.Api/MediaBrowser.Api.csproj]
Playback/Hls/BaseHlsService.cs(213,58): error CS1528: Expected ; or = (cannot specify constructor arguments in declaration) [/home/bond/dev/jellyfin/MediaBrowser.Api/MediaBrowser.Api.csproj]
Playback/Hls/BaseHlsService.cs(213,58): error CS1003: Syntax error, '[' expected [/home/bond/dev/jellyfin/MediaBrowser.Api/MediaBrowser.Api.csproj]
Playback/Hls/BaseHlsService.cs(213,64): error CS1003: Syntax error, ']' expected [/home/bond/dev/jellyfin/MediaBrowser.Api/MediaBrowser.Api.csproj]
```

Reply:
Sorry it would be like this
```cs
var fileStream = GetPlaylistFileStream(playlist);
await using (fileStream.ConfigureAwait(false))
{ 
   ...
}
```

If you want to omit the braces/indent I think you would have to write it like this (the `_` isn't a true discard in this case):

```cs
var fileStream = GetPlaylistFileStream(playlist);
await using var _ = (fileStream.ConfigureAwait(false));
```

Reply:
This new syntax is weird :/

Reply:
I think the first form is really the only one that should be used, but yeah, not a fan :/ It also goes against best practices, because the disposable object is still in scope after the closing brace for the `using`

Reply:
Exactly, I want to use async disposing, but it's just a pain to use

Reply:
Something like this could work too. Create an extra wrapping scope around the fileStream so it still goes out of scope after being disposed.

```
{
   var fileStream = GetPlaylistFileStream(playlist);
   await using var fileStreamAwaiter = fileStream.ConfigureAwait(false);
   ...
}
```